crates/matrix-sdk-ui/src/timeline/event_item/mod.rs

```diff
@@ -67,6 +67,8 @@ pub struct EventTimelineItem {
     pub(super) timestamp: MilliSecondsSinceUnixEpoch,
     /// The content of the event.
     pub(super) content: TimelineItemContent,
+    // The event's encryption information, if any
+    pub(super) encryption_info: Option<EncryptionInfo>,
```

Comment:
Adding this new field is wrong:

- it makes the `encryption_info` stored in the `RemoteEventTimelineItem` redundant, since it's now in two places
- this field only makes sense in the context of remote events, so it should remain in the `RemoteEventTimelineItem`, and should not be added here.

As a matter of fact, I think that most changes in this PR (except for the test, and the other place I've commented) could/should be reverted, unless I missed other calls to `with_encryption_info`.